assets/src/edit-story/components/inspector/document/test/status.js

```diff
@@ -0,0 +1,120 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react';
+import { ThemeProvider } from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import StoryContext from '../../../../app/story/context';
+import InspectorContext from '../../../inspector/context';
+import theme from '../../../../theme';
+import StatusPanel from '../status';
+
+function setupPanel(
+  capabilities = {
+    hasPublishAction: true,
+  }
+) {
+  const updateStory = jest.fn();
+  const deleteStory = jest.fn();
+  const loadStatuses = jest.fn();
+  const loadUsers = jest.fn();
+
+  const storyContextValue = {
+    state: {
+      story: { status: 'draft', password: '' },
+      capabilities,
+    },
+    actions: { updateStory, deleteStory },
+  };
+  const statuses = [
+    {
+      value: 'draft',
+      name: 'Draft',
+    },
+    {
+      value: 'publish',
+      name: 'Public',
+    },
+    {
+      value: 'private',
+      name: 'Private',
+    },
+  ];
+  const inspectorContextValue = {
+    actions: { loadStatuses, loadUsers },
+    state: { statuses },
+  };
+  const { getByText, queryByText } = render(
+    <ThemeProvider theme={theme}>
+      <StoryContext.Provider value={storyContextValue}>
+        <InspectorContext.Provider value={inspectorContextValue}>
+          <StatusPanel />
+        </InspectorContext.Provider>
+      </StoryContext.Provider>
+    </ThemeProvider>
+  );
+  return {
+    getByText,
+    queryByText,
+    updateStory,
+    deleteStory,
+  };
+}
+
+describe('StatusPanel', () => {
+  it('should render Status Panel', () => {
+    const { getByText } = setupPanel();
+    const element = getByText('Status & Visibility');
+    expect(element).toBeDefined();
+
+    const radioOption = getByText('Draft');
+    expect(radioOption).toBeDefined();
+  });
+
+  it('should not render the status option without correct permissions', () => {
+    const { queryByText } = setupPanel({
+      hasPublishAction: false,
+    });
+    expect(queryByText('Draft')).toBeNull();
+  });
+
+  it('should update the story when clicking on status', () => {
+    const { getByText, updateStory } = setupPanel();
+    const publishOption = getByText(/Public/i).closest('label');
+    fireEvent.click(publishOption);
+    expect(updateStory).toHaveBeenCalledWith({
+      properties: {
+        status: 'publish',
+        password: '',
+      },
+    });
+  });
+
+  it('should trigger deleting the story when clicking on delete button', () => {
+    const { deleteStory, queryByText } = setupPanel();
+
+    const deleteButton = queryByText('Move to trash');
+    expect(deleteButton).toBeDefined();
+
+    fireEvent.click(deleteButton);
+    expect(deleteStory).toHaveBeenCalledTimes(1);
```

Comment:
I think it's okay to test it's been called without arguments too:

```suggestion
    expect(deleteStory).toHaveBeenCalledWith();
```

Reply:
It is called with the event object though.